src/main/java/com/juniorstart/juniorstart/service/EmploymentHistoryService.java

```diff
@@ -0,0 +1,83 @@
+import com.juniorstart.juniorstart.model.EmploymentHistory;
+import com.juniorstart.juniorstart.model.UserProfile;
+import com.juniorstart.juniorstart.repository.UserProfileRepository;
+import com.juniorstart.juniorstart.repository.EmploymentHistoryRepository;
+import com.juniorstart.juniorstart.security.UserPrincipal;
+import groovy.util.logging.Slf4j;
+import net.bytebuddy.pool.TypePool;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+@Slf4j
+public class EmploymentHistoryService {
+
+    final private EmploymentHistoryRepository employmentHistoryRepository;
+    final private UserProfileRepository userProfileRepository;
+
+    EmploymentHistoryService(EmploymentHistoryRepository employmentHistoryRepository, UserProfileRepository userProfileRepository) {
+        this.userProfileRepository = userProfileRepository;
+        this.employmentHistoryRepository = employmentHistoryRepository;
+    }
+
+    public ResponseEntity<?> getEmploymentHistory(UUID id, UserPrincipal currentUser) {
+
+        findUser(currentUser.getId());
+
+        return ResponseEntity.ok(userProfileRepository.findByPrivateId(id));
+    }
+
+    public ResponseEntity<?> addEmploymentHistory(EmploymentHistory employmentHistory, UserPrincipal currentUser) {
+
+        Optional<UserProfile> foundUser = findUser(currentUser.getId());
+        checkIsCurrentEmployer(employmentHistory);
+
+        foundUser.get().addEmploymentHistory(employmentHistory);
+
+        return ResponseEntity.ok(userProfileRepository.save(foundUser.get()));
+    }
+
+    public ResponseEntity<?> updateEmploymentHistory(EmploymentHistory employmentHistory, UserPrincipal currentUser) {
+
+        Optional<UserProfile> foundUser = findUser(currentUser.getId());
+        checkIsCurrentEmployer(employmentHistory);
+
+        foundUser.get().getEmploymentsHistory().set(employmentHistory.getId(), employmentHistory);
+
+        foundUser.get().addEmploymentHistory(employmentHistory);
+
+        return ResponseEntity.ok(userProfileRepository.save(foundUser.get()));
+    }
+
+    public ResponseEntity<?> deleteEmploymentHistory(EmploymentHistory employmentHistory, UserPrincipal currentUser) {
+
+        Optional<UserProfile> foundUser = findUser(currentUser.getId());
+
+        if (foundUser.get().getEmploymentsHistory().contains(employmentHistory)) {
+        employmentHistoryRepository.deleteById((long) employmentHistory.getId());
+        return ResponseEntity.ok().build();
+        } else {
+            throw new ResourceNotFoundException("EmploymentHistory", "ID", employmentHistory.getId());
+        }
+    }
+
+    public Optional<UserProfile> findUser(UUID id) {
+        return Optional.ofNullable(userProfileRepository.findByPrivateId(id).orElseThrow(()->
+                new ResourceNotFoundException("UserProfile", "ID", id)));
+    }
```

Comment:
to samo co w innym servisie

Reply:
Tak, tylko tutaj potrzebuję jeszcze zwrócić tą encję, żeby po modyfikacji zapisać ją w bazie.